os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +253,16 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		tcb->stack_guard = (uint32_t)tcb->stack_alloc_ptr + 0x40;
```

Comment:
I am not sure why we need to do `tcb->stack_alloc_ptr + 0x40`.
But, I have an alternative suggestion to have 32 byte aligned address. We can use memalign instead of malloc as follows:

`tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(32, stack_size);`

Reply:
Modified to reserve memory with kmm_memalign. After this the size will be fixed to 32 byte now.